File: java/Easy/IsSubsequence.java
public class IsSubsequence {

    public static void main(String[] args){
        String a = "acc";
        String b = "ahbcgdbc";
        System.out.println(isSubsequence(a,b));
     
    }
    private static boolean isSubsequence(String s, String t) {
        int found = 0;
        int jIndx = 0;
        for (int i = 0; i < s.length(); i ++){
            for (int j = jIndx; j < t.length(); j++){
                if (s.charAt(i)==t.charAt(j)){
                    found++;
                    jIndx = j+1;
                    break;
                }
            }
        }
        if (s.length()==found){
            return true;
        } else {
            return false;
        }
    }
}

File: kotlin/Easy/IsSubsequence.kt
package Easy

fun main() {
    var a : String = "acc"
    var b : String = "ahbccgdb"
    println(isSubsequence(a ,b))
    println(isSubsequence2(a ,b))
}

fun isSubsequence(s: String, t: String): Boolean {
    var found = 0
    var jIndx = 0
    for (i in s.indices){
        for(j  in jIndx..t.length - 1){
            if(s[i] == t[j]){
                found++
                jIndx = j + 1;
                break;
            }
        }
    }
    return found == s.length;
}

fun isSubsequence2(s: String, t: String): Boolean {
    var sch = 0
    var tch = 0
    while(sch < s.length && tch < t.length){
        if(s[sch] == t[tch]) sch++
        tch++
    }
    return sch == s.length
}